gcc/rust/resolve/rust-name-resolver.cc

```diff
@@ -0,0 +1,37 @@
+#include <rust-name-resolver.h>
+
+namespace Rust {
+namespace Resolver {
+
+  void Scope::append_reference_for_def (NodeId refId, NodeId defId)
+  {
```

Comment:
I like the move to pull things into implementation files. Just remember to add the GCC Copyright header too